From 9e7b9f2f4b03b376b185fb2c0dc3428ae2bf00bc Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Fri, 4 Oct 2013 13:12:24 +0600
Subject: [PATCH] Fixed Toolbar Options again

Options internally uses QHash so they has no order.
Before putting toolbars in combobox they must be sorted.

The patch wan't applyed by upstream. So I commit here.
---
 src/options/opt_toolbars.cpp | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/options/opt_toolbars.cpp b/src/options/opt_toolbars.cpp
index 48d0f1e..8109e04 100644
--- a/src/options/opt_toolbars.cpp
+++ b/src/options/opt_toolbars.cpp
@@ -22,6 +22,7 @@
 #include <QHBoxLayout>
 #include <QList>
 #include <QVBoxLayout>
+#include <QRegExp>
 
 #define CHAT_TOOLBAR       0
 #define GROUPCHAT_TOOLBAR  1
@@ -222,10 +223,20 @@ void OptionsTabToolbars::restoreOptions()
 	QStringList toolbarBases;
 	toolbarBases = o->getChildOptionNames("options.ui.contactlist.toolbars", true, true);
 
+
+	// toolbarBases is chaotic. Need to sort before do anything.
+	QStringList sortedToolbarBases;
+	for (int i = 0; !toolbarBases.isEmpty(); ++i) {
+		int index = toolbarBases.indexOf(QRegExp(QString(".*m%1$").arg(i)));
+		if (index >= 0) {
+			sortedToolbarBases << toolbarBases.takeAt(index);
+		}
+	}
+
 	QString chatToolbarName = tr("Chat");
 	QString groupchatToolbarName = tr("Groupchat");
 
-	foreach(QString base, toolbarBases) {
+	foreach(QString base, sortedToolbarBases) {
 		ToolbarPrefs tb;
 		tb.id = o->getOption(base + ".key").toString();
 		tb.name = o->getOption(base + ".name").toString();
-- 
1.8.3.1

